python/scripts/merge_vcf_MNVs.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python
+#   LICENSE
+# Copyright (c) 2021 Genome Research Ltd.
+# Author: Cancer Genome Project cgphelp@sanger.ac.uk
+#
+#
+# This file is part of CASM-Smart-Phase.
+#
+# CASM-Smart-Phase is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the Free
+# Software Foundation; either version 3 of the License, or (at your option) any
+# later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+#    1. The usage of a range of years within a copyright statement contained within
+#    this distribution should be interpreted as being equivalent to a list of years
+#    including the first and last year specified and all consecutive years between
+#    them. For example, a copyright statement that reads ‘Copyright (c) 2005, 2007-
+#    2009, 2011-2012’ should be interpreted as being identical to a statement that
+#    reads ‘Copyright (c) 2005, 2007, 2008, 2009, 2011, 2012’ and a copyright
+#    statement that reads ‘Copyright (c) 2005-2012’ should be interpreted as being
+#    identical to a statement that reads ‘Copyright (c) 2005, 2006, 2007, 2008,
+#    2009, 2010, 2011, 2012’."
+#
+#
+"""
+merge_vcf_MNVs.py: Script to merge adjacent SNVs in a CaVEMan generated
+VCF into a new VCF containing SNVs and merged MNVs in order to be
+processed by Smart-phase
+"""
+# Core imports
+import argparse
+import os
+import sys
+
+import pkg_resources  # part of setuptools
+from casmsmartphase.MNVMerge import MNVMerge
+
+VERSION = pkg_resources.require("casmsmartphase")[0].version
+
+
+# Define arguments and argument associated methods
+def generate_arg_str(parsed_args):
+
+    ag_str = ""
+    idx = 0
+    for arg in sorted(parsed_args.__dict__.keys()):
+        val = parsed_args.__dict__[arg]
+        if not val:
+            continue
+        if arg == "vcfin" or arg == "vcfout" or arg == "bedfile":
+            val = os.path.basename(val)
+
+        if arg == "flagremove":
+            inner_idx = idx
+            for v in val:
+                if inner_idx > 0:
+                    ag_str += ","
+                ag_str += "{}={}".format(arg, v)
+                inner_idx += 1
+        else:
+            if idx > 0:
+                ag_str += ","
+            ag_str += "{}={}".format(arg, val)
+        idx += 1
```

Comment:
Should you be capturing what the user provided rather than the parsed arguments?  Can cause issues if you find the defaults are being applied incorrectly as the header will be indicating the user specified items they did not.

Capturing what the user "did" and just appending the defaults as a supplementary block something like the following may be better:

```python3
command = ""
for i, e in enumerate(sys.argv):
    if i == 0:
        command += f"{os.path.basename(e)}"
        continue
    command += f" {e}"
command += f" (defaults: {DEFAULT_PARAMS})"
```

Reply:
Since changing to click cli inputs this has been moved to cli.py. All defaults are at the end of the string.